ecosystem/sep-0031.md

```diff
@@ -249,6 +268,51 @@ In the case where the transaction just cannot be completed, return an error resp
 }
 ```
 
+### Fee
+
+The fee endpoint allows an receiving anchor to report the fee that would be charged for processing the payment of an asset. This endpoint is important to allow a sending anchor to accurately report fees to a sending user even when the fee schedule is complex. If a fee can be fully expressed with the `fee_fixed` and `fee_percent` fields in the `/info` response, then an anchor should not implement this endpoint.
+
+```
+GET DIRECT_PAYMENT_SERVER/fee
+```
+
+Request parameters:
+
+Name | Type | Description
+-----|------|------------
+`type` | string | (optional) Type of payment method, optionally described in the `GET /info` endpoint
+`asset_code` | string | Asset code of the payment
+`amount` | float | Amount of the asset that would be sent to the receiving user
```

Comment:
❗ I think we should make the `amount` request parameter a string, and note in the description it should be a decimal. That would be consistent with the other amount fields that already exist in SEP-31, for e.g.:
https://github.com/stellar/stellar-protocol/blob/e195a267aa3f2c27a78c12e85676ed33a806debb/ecosystem/sep-0031.md#L303-L305

Why: The consistency will make it less taxing on the developer consuming the API if they can assume amounts are represented and consumed in the same way everywhere. A 'float' comes with some baggage. Floats can be represented with an exponent sometimes by some programming languages when serializing and I don't think we'd want to complicate parsing. Also some languages have types that are much better at naturally handling money than floats.

```suggestion
`amount` | string | Amount of the asset that would be sent to the receiving user as a string with up to 7 decimals.
```

Would 7 decimals suffice?